sympy/plotting/plot.py

```diff
@@ -1044,47 +1168,64 @@ def process_series(self):
+            if isinstance(ax, Axes3D):
                 pass
             elif val == 'center':
-                self.ax.spines['left'].set_position('center')
-                self.ax.spines['bottom'].set_position('center')
+                ax.spines['left'].set_position('center')
+                ax.spines['bottom'].set_position('center')
             elif val == 'auto':
-                xl, xh = self.ax.get_xlim()
-                yl, yh = self.ax.get_ylim()
+                xl, xh = ax.get_xlim()
+                yl, yh = ax.get_ylim()
                 pos_left = ('data', 0) if xl*xh <= 0 else 'center'
                 pos_bottom = ('data', 0) if yl*yh <= 0 else 'center'
-                self.ax.spines['left'].set_position(pos_left)
-                self.ax.spines['bottom'].set_position(pos_bottom)
+                ax.spines['left'].set_position(pos_left)
+                ax.spines['bottom'].set_position(pos_bottom)
             else:
-                self.ax.spines['left'].set_position(('data', val[0]))
-                self.ax.spines['bottom'].set_position(('data', val[1]))
+                ax.spines['left'].set_position(('data', val[0]))
+                ax.spines['bottom'].set_position(('data', val[1]))
         if not parent.axis:
-            self.ax.set_axis_off()
+            ax.set_axis_off()
         if parent.legend:
-            if self.ax.legend():
-                self.ax.legend_.set_visible(parent.legend)
+            if ax.legend():
+                ax.legend_.set_visible(parent.legend)
         if parent.margin:
-            self.ax.set_xmargin(parent.margin)
-            self.ax.set_ymargin(parent.margin)
+            ax.set_xmargin(parent.margin)
+            ax.set_ymargin(parent.margin)
         if parent.title:
-            self.ax.set_title(parent.title)
+            ax.set_title(parent.title)
         if parent.xlabel:
-            self.ax.set_xlabel(parent.xlabel, position=(1, 0))
+            ax.set_xlabel(parent.xlabel, position=(1, 0))
         if parent.ylabel:
-            self.ax.set_ylabel(parent.ylabel, position=(0, 1))
+            ax.set_ylabel(parent.ylabel, position=(0, 1))
+
+    def process_series(self):
+        """
+        Iterates over every ``Plot`` object and further calls
+        _process_series()
+        """
+        parent = self.parent
+        if isinstance(parent, Plot):
+            series_list = [parent._series]
+        else:
+            series_list = parent._series
+
+        for i, (series, ax) in enumerate(zip(series_list, self.ax)):
+            if isinstance(self.parent, PlotGrid):
+                parent = self.parent.args[i]
+            self._process_series(series, ax, parent)
 
     def show(self):
         self.process_series()
         #TODO after fixing https://github.com/ipython/ipython/issues/1255
         # you can uncomment the next line and remove the pyplot.show() call
         #self.fig.show()
         if _show:
+            self.fig.tight_layout()
```

Comment:
I don't think this should be set here, it changes the expected output. Please make this an option or make it only work for PlotGrid.

Reply:
As a very minor counterpoint (feel free to disregard it): I **personally** think `tight_layout` provides some nice defaults. Maybe it clashes with some "mathematical plots" conventions which would indeed be bad, but for the most part is seems nice (in single plots and in plot grids).

Reply:
My only concern is that tight_layout has never been called before, so it may change the expected output of all prior plots made with the module. My concern is with the backwards compatibility of the outputs of plot().

Reply:
But, thinking the other way, isn't a single plot object defined as a subplot of grid 1x1?
So if the `tight_layout()` is applied only to `PlotGrid` objects, shouldn't `Plot` objects should also be included with the same?

Reply:
Yes, tight_layout() is likely preferred for all plots, but if this changes the currently expected output is a backwards incompatible break. That is my only concern about this. Does it change the output of a single plot at all? If so, do we want to change this functionality without a deprecation? etc.

Reply:
Then, I think a `tight_layout` kwarg could be added to give an option to the user, similar to the `show` kwarg.
Please suggest.

Reply:
I can't make a suggestion until you let me know if tight_layout() changes all existing plots. Have you checked all the plots in the documentation to see if your changes affect them?

Reply:
I had checked them before `tight_layout()` was introduced.
So I will be checking them again and would ping you once its done.

Reply:
If you have already checked, then can you answer whether it affects the output or not? That is all I want to know.

Reply:
> Then, I think a `tight_layout` kwarg could be added to give an option to the user, similar to the `show` kwarg.

I do not think this is a good idea: `tight_layout` is a matplotlib specific bandaid that should not be visible to the user, especially given that it does not make sense for other plotting backends.

Reply:
I have checked the examples and there seems to be no difference from the old plots, apart from a bit better spacing in the new one.

Reply:
Maybe you can show us the differences you are talking about?

Reply:
One of the example that I tried was the 3d parametric line given [here](https://github.com/sympy/sympy/blob/d383496deb5bed647ed930fd0194c0f57dba495d/examples/beginner/plot_colors.ipynb) and got this:
The left one is the output from this branch and the right one the output of the old branch

![Screenshot - 26-03-2019 , 23_51_07](https://user-images.githubusercontent.com/29159034/55031110-d9d9e200-5033-11e9-9992-81b8836291ed.png)

I don't know wether the difference is visible to you or not, but the left one(of this branch) is a bit 
bigger and the tick labels of the axis are also better spaced.
The other results also had similar differences
Not sure wether these should be counted as differneces.

Reply:
Ok, I'm fine with this change.